tests/components/evohome/conftest.py

```diff
@@ -20,79 +22,121 @@
+            or self.access_token_expires < datetime.now()
+        ):
+            self.access_token = f"new_{ACCESS_TOKEN}"
+            self.access_token_expires = datetime.now() + timedelta(minutes=30)
 
-    if self.access_token_expires is None or self.access_token_expires < datetime.now():
-        self.access_token = f"new_{ACCESS_TOKEN}"
-        self.access_token_expires = datetime.now() + timedelta(minutes=30)
+        # assume a valid GET, and return the JSON for that web API
+        if url == "userAccount":  # userAccount
+            return user_account_config_fixture(install)
 
-    # assume a valid GET, and return the JSON for that web API
-    if url == "userAccount":  #                    userAccount
-        return user_account_config_fixture()
+        if url.startswith("location"):
+            if "installationInfo" in url:  # location/installationInfo?userId={id}
+                return user_locations_config_fixture(install)
+            if "location" in url:  # location/{id}/status
+                return location_status_fixture(install)
 
-    if url.startswith("location"):
-        if "installationInfo" in url:  #           location/installationInfo?userId={id}
-            return user_locations_config_fixture()
-        if "location" in url:  #                   location/{id}/status
-            return location_status_fixture("2738909")
+        elif "schedule" in url:
+            if url.startswith("domesticHotWater"):  # domesticHotWater/{id}/schedule
+                return dhw_schedule_fixture(install)
+            if url.startswith("temperatureZone"):  # temperatureZone/{id}/schedule
+                return zone_schedule_fixture(install)
 
-    elif "schedule" in url:
-        if url.startswith("domesticHotWater"):  #  domesticHotWater/{id}/schedule
-            return dhw_schedule_fixture()
-        if url.startswith("temperatureZone"):  #   temperatureZone/{id}/schedule
-            return zone_schedule_fixture()
+        pytest.fail(f"Unexpected request: {HTTPMethod.GET} {url}")
 
-    pytest.xfail(f"Unexpected URL: {url}")
+    return mock_get
 
 
-@patch("evohomeasync2.broker.Broker.get", mock_get)
-async def setup_evohome(hass: HomeAssistant, test_config: dict[str, str]) -> MagicMock:
+async def block_request(
+    self: Broker, method: HTTPMethod, url: str, **kwargs: Any
+) -> None:
+    """Fail if the code attempts any actual I/O via aiohttp."""
+
+    pytest.fail(f"Unexpected request: {method} {url}")
+
+
+@patch("evohomeasync.broker.Broker._make_request", block_request)
+@patch("evohomeasync2.broker.Broker._client", block_request)
+async def setup_evohome(
+    hass: HomeAssistant,
+    test_config: dict[str, str] | None = None,
+    install: str = "default",
+) -> MagicMock:
     """Set up the evohome integration and return its client.
 
     The class is mocked here to check the client was instantiated with the correct args.
     """
 
+    if test_config is None:
```

Comment:
Make the test config a pytest fixture instead and return the default config. Then we can override the fixture and default config via `pytest.mark.parametrize` as needed in the tests.

Try to avoid complicated logic with many conditions also in the test helpers.